### PR TITLE
Bump the Pulsar reactive client to 0.2.0

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java [hunk 1]
```diff
@@ -137,13 +140,14 @@ private Mono<MessageId> doSend(@Nullable String topic, T message, @Nullable Sche
...
+	private Flux<MessageSendResult<T>> doSendMany(@Nullable String topic, Flux<MessageSpec<T>> messages,
+			@Nullable Schema<T> schema) {
 		return messages.switchOnFirst((firstSignal, messageFlux) -> {
-			T firstMessage = firstSignal.get();
+			MessageSpec<T> firstMessage = firstSignal.get();
 			if (firstMessage != null && firstSignal.isOnNext()) {
-				String topicName = resolveTopic(topic, firstMessage.getClass());
-				ReactiveMessageSender<T> sender = createMessageSender(topicName, firstMessage, schema, null);
-				return messageFlux.map(MessageSpec::of).as(sender::sendMany)
+				String topicName = resolveTopic(topic, firstMessage.getValue().getClass());
```
Comment (cbornet) [inline spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java:148]: The good news is that with `MessageSpec` we can send tombstone/null values (it's not possible with a Flux<T> as a Flux can't emit null.
The bad news is that if the first message is a tombstone, we can't determine the topic or the schema (and this code will NPE 😬

File: build.gradle [hunk 2]
```diff
@@ -24,6 +24,12 @@ allprojects {
 	configurations.all {
 		resolutionStrategy.cacheChangingModulesFor 0, "minutes"
 	}
+
+	repositories {
```
Comment (onobc) [inline build.gradle:28]: This is a temporary override of the regular snapshot repo (defined [here](https://github.com/spring-projects-experimental/spring-pulsar/blob/b9710b947a258925e80a42393a0976a07a4327a7/buildSrc/src/main/groovy/io/spring/gradle/convention/RepositoryConventionPlugin.groovy#L52)) and will be removed once `0.2.0` is released, correct? This allows us to move forward and get a green CI run now.

File: spring-pulsar-docs/src/main/asciidoc/reactive-quick-tour.adoc [hunk 3]
```diff
@@ -45,7 +45,7 @@ public class ReactiveSpringPulsarHelloWorld {
 
     @Bean
     ApplicationRunner runner(ReactivePulsarTemplate<String> pulsarTemplate) {
-        return (args) -> pulsarTemplate.send("hello-pulsar-topic", "Hello Reactive Pulsar World!").subscribe();
+        return (args) -> pulsarTemplate.send("hello-pulsar-topic", Flux.just(MessageSpec.of("Hello Reactive Pulsar World!"))).subscribe();
```
Comment (onobc) [inline spring-pulsar-docs/src/main/asciidoc/reactive-quick-tour.adoc:48]: I think the previous sample use w/ single message may be less to digest on the quick-tour.

File: spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java [hunk 4]
```diff
@@ -62,9 +63,10 @@ public ReactivePulsarTemplate(ReactivePulsarSenderFactory<T> reactiveMessageSend
 	}
 
 	/**
-	 * Construct a template instance with a custom schema resolver.
+	 * Construct a template instance with a custom schema resolver and a custom topic
```
Comment (onobc) [inline spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java:66]: Good catch - thx for the update.

File: spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java (hunk 1 above)
Comment (onobc) [reply spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java:148]: I like the good news better! 🤣 That is great news w/ the tombstone ability. 

I feel like we should add a null check on the value and log something accordingly? 

So basically we support tombstone now w/ send many w/ the one caveat that they can not be the 1st message in the published stream. Can we "switchOnSecond" in that case? I don't know if that is even a thing - just thinking of ideas Even if that worked, could they have 2 null values back to back at position 1 and 2?

**Review by onobc — CHANGES_REQUESTED** [review]
@cbornet good call on pre-loading this PR w/ the changes to the builder - it made this much easier to digest. 

LGTM! I added a couple of comments. After we resolve those then this is good to go.

File: spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java (hunk 1 above)
Comment (cbornet) [reply spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java:148]: > Even if that worked, could they have 2 null values back to back at position 1 and 2?

They could have an infinite stream of nulls 😉

File: build.gradle (hunk 2 above)
Comment (cbornet) [reply build.gradle:28]: Exactly. That's why the PR is in Draft mode atm

File: spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java (hunk 1 above)
Comment (cbornet) [reply spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java:148]: Or a Flux that emits just one null MS and completes.

Comment (cbornet) [reply spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java:148]: BTW we have the same problem for the `send(T message)` operation.

Comment (onobc) [reply spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java:148]: True and true. 
This PR is already above/beyond its initial goal. I have created https://github.com/spring-projects-experimental/spring-pulsar/issues/311 to follow up w/ the docs for sending tombstones.

File: build.gradle [hunk 5]
```diff
@@ -24,6 +24,12 @@ allprojects {
...
+		maven {
```
Comment (onobc) [inline build.gradle:29]: Looks like we can pull this now @cbornet 

https://repo.maven.apache.org/maven2/org/apache/pulsar/pulsar-client-reactive-api/0.2.0/

**Review by onobc — CHANGES_REQUESTED** [review]
https://github.com/spring-projects-experimental/spring-pulsar/pull/307/files#r1098152729

File: build.gradle (hunk 5 above)
Comment (cbornet) [reply build.gradle]: Indeed. Done.

**Review by onobc — APPROVED** [review]
@cbornet - lets go!

**Pull request merged**